homeassistant/components/tradfri/sensor.py

```diff
@@ -48,18 +63,41 @@ def _get_air_quality(device: Device) -> int | None:
     return cast(int, device.air_purifier_control.air_purifiers[0].air_quality)
 
 
-SENSOR_DESCRIPTION_AQI = TradfriSensorEntityDescription(
-    device_class=SensorDeviceClass.AQI,
-    native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
-    key=SensorDeviceClass.AQI,
-    value=_get_air_quality,
+def _get_filter_time_left(device: Device) -> int:
+    """Fetch the filter's remaining life (in hours)."""
+    return cast(
+        int, device.air_purifier_control.air_purifiers[0].filter_lifetime_remaining
```

Comment:
Why does the hours show as a decimal number in PR description screen shot?

Reply:
Because I can't read my own docstrings 😊 The API returns in minutes. Does it make more sense to just expose that or to expose in hours as suggested here?

Reply:
(I'm using comma as thousans delimiter so it's 105 600 minutes left).

Reply:
I think we should round it to hours. Minutes should never be interesting.

Have we checked that the value only changes when the device is in use, and not just because time passes?

Are there separate state change events for this feature? Does it update every minute when running the device?

Reply:
No, let me check how it works and revert.

Reply:
wouldn't it be even better to recalculate that to days/months, if over 24 hours... just like the app?

Reply:
![20220206_135117000_iOS](https://user-images.githubusercontent.com/33354141/152684105-8d4b46e4-ed96-4d7e-80e6-8a367958477b.jpg)

Reply:
@Mariusthvdb let's leave that to the user for a template sensor.

Reply:
It doesn't seem to update every minute, let me see if I can track how often it updates.

Reply:
Just noted an update and subscription seems to work as only `5910` updated:
```
Initial state:
{'5900': 3, '5902': 153945, '5903': 0, '5904': 259200, '5905': 0, '5906': 1, '5908': 10, '5907': 9, '5909': 153984, '5910': 105255, '9003': 0}

Changed state:
{'5900': 3, '5902': 153995, '5903': 0, '5904': 259200, '5905': 0, '5906': 1, '5908': 10, '5907': 9, '5909': 153984, '5910': 105205, '9003': 0}
```
https://github.com/home-assistant-libs/pytradfri/blob/4ec16dc5f0f2ae5fe5858e59352632583c2d6af2/pytradfri/device/air_purifier.py#L41

The fan is currently on. I'll leave this running for a while so we can see what happens.

Reply:
It seems to have a cadence of 50 minutes.

Reply:
Agree. Let's have a look in about 30 minutes. I'll turn on the fan then and see what it returns.

Reply:
Very odd, it's decreased by 15 minutes.

```
{'5900': 1, '5902': 154060, '5903': 0, '5904': 259200, '5905': 0, '5906': 1, '5908': 30, '5907': 65535, '5909': 154044, '5910': 105140, '9003': 0}
```

Reply:
Another check now. It's been off for 90 minutes and it's decreased by 12 minutes. Seems like it's decaying slowly when off and then by the minute when it's on.
```
{'5900': 0, '5902': 154072, '5903': 0, '5904': 259200, '5905': 0, '5906': 1, '5908': 0, '5907': 65535, '5909': 154044, '5910': 105128, '9003': 0}
```